### PR TITLE
Charlesmchen/signal account

### PR DESCRIPTION
**Pull request by charlesmchen**
* Begin transition to new `SignalAccount` class.
* Extract labels for phone numbers.

PTAL @michaelkirk

### CODE REVIEW

File: src/Contacts/Contact.h [hunk 1]
```diff
@@ -32,18 +48,23 @@ NS_ASSUME_NONNULL_BEGIN
 #endif // TARGET_OS_IOS
 
 - (BOOL)isSignalContact;
+- (NSArray<SignalRecipient *> *)signalRecipients;
+// TODO: Remove this method.
```
Comment (michaelkirk) [inline src/Contacts/Contact.h]: :+1:

File: src/Contacts/Contact.h [hunk 2]
```diff
@@ -1,8 +1,23 @@
...
 #import <Foundation/Foundation.h>
 
 NS_ASSUME_NONNULL_BEGIN
 
+typedef NS_ENUM(NSUInteger, OWSPhoneNumberType) {
+    OWSPhoneNumberTypeUnknown,
+    OWSPhoneNumberTypeMobile,
+    OWSPhoneNumberTypeIPhone,
+    OWSPhoneNumberTypeMain,
+    OWSPhoneNumberTypeHomeFAX,
```
Comment (michaelkirk) [inline src/Contacts/Contact.h:15]: nit: "FAX" -> "Fax"

**Review by michaelkirk — APPROVED** [review]
LGTM.

Intuitively, I feel like "phoneNumberType" should live on the PhoneNumber class, rather than having this  separate phone number type map on the contact, but that might require some heavy handed changes.

**Comment by charlesmchen** [comment]
PTAL @michaelkirk I've added one more commit here:

https://github.com/WhisperSystems/SignalServiceKit/pull/194/commits/5058eb837ea2e33a486a460b16f5a797241d4c76

**Comment by michaelkirk** [comment]
LGTM

File: src/Contacts/Contact.h (hunk 2 above)
Comment (michaelkirk) [reply src/Contacts/Contact.h:15]: I take it back, I guess you were just following the ABConvention: `kABPersonPhoneHomeFAXLabel`

**Pull request merged**